### DISCUSSION TITLE
VSCode extension frontend

### DISCUSSION THREAD

**Discussion by IgorAndrade10** [FEUP-MEIC-DS-2024-25/ai4sd#12] [Ideas]
Hello everyone! After completing the prototype for sprint 0, my team already has the code for a basic VSCode extension that, with some tweaks, could be used as the base code for the frontend of the VSCode extensions of the whole project.
The language chosen was TypeScript, mainly since it was the recommended language by the VSCode team itself, as the VSCode extension API itself is written in TypeScript.
If anyone has any better ideas, a better start code or is completly against this please propose it in this discussion. I'll wait on responses before coming to any conclusion.

**Comment by Onso37** [comment]
I'm not against the idea of having some shared base component between extensions, but I don't think bundling all the assistants into a single extension is the best idea. A more modular version of your idea, I think, would be to make the frontend its own extension, that the other assistants would depend on. Instead of each assistant having its code files inside the common extension, the base extension would expose a function that allows the assistants to register themselves to the dropdown menu (which wouldn't be too different to what you have now). This way, the user could pick which assistants to install (and it would leave the door open for the addition of more assistants without changing the base code).

**Comment by aaguiar** [comment]
Hi, considering VSC extensions, I am sure that either only one extension for the whole AI4SD or one extension per team is not a good idea. Maybe a number between 1 and 33 will be the optimum. Let's wait for the eventual merge of assistants to get stable and then we decide if 1, 33, or 10, or 20 VSC extensions.

But this is clearly a topic for discussion. Let's keep it open.
My 2 cents.
